neuro_san/service/generic/async_agent_service.py

```diff
@@ -241,6 +243,12 @@ async def streaming_chat(self, request_dict: Dict[str, Any],
         request_reporting: Dict[str, Any] = invocation_context.get_request_reporting()
         invocation_context.close()
 
+        # Maybe report token accounting to a UsageLogger
+        token_dict: Dict[str, Any] = request_reporting.get("token_accounting")
+        if token_dict is not None:
+            usage_logger: WrappedUsageLogger = UsageLoggerFactory.create_usage_logger()
+            await usage_logger.log_usage(token_dict, request_metadata)
```

Comment:
Have the asynchronous service employ the UsageLogger via the new infrastructure.